Workaround: just skip anonymous superclasses and included modules

tadd opened the pull request:
Fixes #301 and https://github.com/ruby/rbs/issues/301#issuecomment-641747375 based on the direction of https://github.com/ruby/rbs/issues/301#issuecomment-641726359.

For this `./c.rb`,

```ruby
class C < Class.new
  include Module.new
  def x; end
end
```

the command `bundle exec rbs prototype runtime --require ./c C` generates warning messages and  signatures:

```
W, [2020-06-10T15:39:12.732355 #28227]  WARN -- rbs: Skipping anonymous superclass #<Class:0x000055b123e491a0>
W, [2020-06-10T15:39:12.732475 #28227]  WARN -- rbs: Skipping anonymous module #<Module:0x000055b123e49010>
class C
  public

  def x: () -> untyped
end
```

pocke commented:
I guess `generate_module` method has the same problem (I don't test it).
https://github.com/ruby/rbs/blob/76f6b9eef89a03eea3f5ce46e007bda41c3aef74/lib/rbs/prototype/runtime.rb#L359


```ruby
module M
  include Module.new
end
```

tadd commented:
@pocke Thanks, you're right. I fixed it with 67ecdce.

In addition, though this may be a corner case, I fixed a similar case with 23c3574 for constants.

```
module M
   C = Class.new.new # instance of an anonymous class
end
```

Because it yielded useless output.

```
M::C: #<Class:0x00005642f1feeea0>
```

tadd commented on lib/rbs/prototype/runtime.rb:
```diff
@@ -321,6 +329,11 @@ def generate_class(mod)
+          unless included_module.name
+            RBS.logger.warn("Skipping anonymous module #{included_module}")
+            next
+          end
```
Should I move this and [unless clause below](https://github.com/ruby/rbs/pull/302/files#diff-d15f6899dad56f0e659ff044e5c71b36R377-R381) to the inside of `def each_mixin`?

soutaro replied:
My opinion is to keep the test here.

soutaro commented:
@tadd Could you add tests to confirm that the RBI generation doesn't result in an exception?

tadd commented:
I'm trying to add tests.

tadd commented:
a0e37db: A test added for `decls` method.

At first, I tried to add tests for `generate_{constants,class,module}` separately, but it seemed to require some unnatural things:

* Manipulating instance vars with `p.instance_variable_set(:@decls, [])` before each tests, or
* Changing signatures of `generate_*` to return an Array
  * This may yield (a bit of?) slowdown

So I chose the current style but any comments welcome.

0cc0afa: Warning messages improved. Now warns like

```
Skipping anonymous superclass #<Class:0x00005623fddad4b0> of A::C
Skipping anonymous module #<Module:0x00005623fddad320> included in A::C2
Skipping constant CONST #<#<Class:0x00005623fddad258>:0x00005623fddad168> of A::M as an instance of anonymous class
Skipping anonymous module #<Module:0x00005623fddad028> included in A::M2
```

for a file of

```ruby
module A
  class C < Class.new
  end

  class C2
    include Module.new
  end

  module M
    CONST = Class.new.new
  end

  module M2
    include Module.new
  end
end
```

tadd commented:
@soutaro Thanks. I will squash all commits into one.
